src/utils/daos_hdlr.c

```
@@ -1561,6 +1562,817 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+				rc = -ENOMEM;
+				goto out;
+			}
+		} else {
+			char *new_dir = calloc(strlen(cwd) + strlen(cont_name) + 1, sizeof(char));
+			if (new_dir == NULL) {
+				rc = -ENOMEM;
+				goto out;
+			}
+
+			strcpy(new_dir, cwd);
+			if (cont_name[0] == '.') {
+				strcat(new_dir, &cont_name[1]);
+			} else {
+				strcat(new_dir, "/");
+				strcat(new_dir, cont_name);
+			}
+			cont_name = new_dir;
+		}
+		*_cont_name = cont_name;
+	} else {
+		*_cont_name = strdup(cont_name);
+		if (*_cont_name == NULL) {
+			rc = -ENOMEM;
+			goto out;
+		}
+	}
+	*_obj_name = strdup(fname);
+	if (*_obj_name == NULL) {
+		free(*_cont_name);
+		*_cont_name = NULL;
+		rc = -ENOMEM;
+		goto out;
+	}
+	out:
+		if (f1)
+			free(f1);
+		if (f2)
+			free(f2);
+	return rc;
+}
+
+static ssize_t
+daos_write(struct daos_file_t *daos_file, char *file, void *buf, size_t size)
+{
+	int rc;
+	/* record address and size of user buffer in io vector */
+	d_iov_t iov;
+	d_iov_set(&iov, buf, size);
+
+	/* define scatter-gather list for dfs_write */
+	d_sg_list_t sgl;
+	sgl.sg_nr = 1;
+	sgl.sg_iovs = &iov;
+	sgl.sg_nr_out = 1;
+
+	rc = dfs_write(daos_file->dfs, daos_file->obj, &sgl, daos_file->offset, NULL); 
+
+	if (rc) {
+		fprintf(stderr, "dfs_write %s failed (%d %s)\n",
+			file, rc, strerror(rc));
+		errno = rc;
+		size = -1;
+	}
+
+	daos_file->offset += (daos_off_t)size;
+	return (ssize_t) size;
+}
+
+static ssize_t
+daos_file_write(daos_file_t *daos_file, char* file,
+		void* buf, size_t size)
+{
+	ssize_t num_bytes_written = 0;
+	if (daos_file->type == POSIX) {
+		num_bytes_written = write(daos_file->fd, buf, size);
+	} else if (daos_file->type == DAOS) {
+		num_bytes_written = daos_write(daos_file, file, buf, size);
+	} else {
+		fprintf(stderr, "File type not known: %s type=%d\n",
+			file, daos_file->type);
+	}
+	if (num_bytes_written < 0) {
+		fprintf(stderr, "write error on %s type=%d\n",
+			file, daos_file->type);
+	}
+	return num_bytes_written;
+}
+
+static void 
```

this should return an error, why void?